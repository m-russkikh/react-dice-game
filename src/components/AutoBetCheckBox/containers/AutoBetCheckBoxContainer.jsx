import React, { Component } from 'react';

import CheckBoxControl from '../../CheckBoxControl';
import { AutoBetCheckBoxContainerPropType } from '../propTypes';


class AutoBetCheckBoxContainer extends Component {
  handleChange = ({ currentTarget: { checked } }) => {
    this.props.setAutoBet(checked);
  }

  render() {
    return (
      <CheckBoxControl
        label="Auto Bet"
        onChange={this.handleChange}
        disabled={!this.props.numberOfBets}
      />
    );
  }
}

AutoBetCheckBoxContainer.propTypes = AutoBetCheckBoxContainerPropType;

export default AutoBetCheckBoxContainer;
