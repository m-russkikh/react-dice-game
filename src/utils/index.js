import { MAX_BET_NUMBER } from '../constants/diceGame';


export function saveState(state) {
  const serializedBalance = JSON.stringify(state.balance);

  localStorage.setItem('balance', serializedBalance);
}

export function getBetNumber() {
  return Math.floor(Math.random(0, 1) * MAX_BET_NUMBER) + 1;
}
