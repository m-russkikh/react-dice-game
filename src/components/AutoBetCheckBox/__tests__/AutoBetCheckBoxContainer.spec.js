import { getShallowWrapper } from '../../../utils/testUtils';
import AutoBetCheckBoxContainer from '../containers/AutoBetCheckBoxContainer';


describe('AutoBetCheckBoxContainer', () => {
  let props;
  let wrapper;
  let instance;

  beforeEach(() => {
    props = {
      numberOfBets: 1,
      setAutoBet: jest.fn(),
    };

    wrapper = getShallowWrapper(AutoBetCheckBoxContainer, props);
    instance = wrapper.instance();
  });

  describe('render', () => {
    it('should display the disabled checkbox', () => {
      wrapper.setProps({ ...props, numberOfBets: null });

      expect(wrapper.debug()).toMatchSnapshot();
    });

    it('should display the enabled checkbox', () => {
      wrapper.setProps({ ...props, numberOfBets: 1 });

      expect(wrapper.debug()).toMatchSnapshot();
    });
  });

  describe('methods', () => {
    describe('handleChange', () => {
      it('should call setAutoBet', () => {
        const event = { currentTarget: { checked: true } };

        instance.handleChange(event);

        expect(props.setAutoBet).toHaveBeenCalledTimes(1);
        expect(props.setAutoBet).toHaveBeenCalledWith(event.currentTarget.checked);
      });
    });
  });
});
