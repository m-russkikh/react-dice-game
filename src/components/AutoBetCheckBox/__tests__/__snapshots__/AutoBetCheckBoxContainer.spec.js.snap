// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AutoBetCheckBoxContainer render should display the disabled checkbox 1`] = `"<CheckBoxControlView label=\\"Auto Bet\\" onChange={[Function]} disabled={true} />"`;

exports[`AutoBetCheckBoxContainer render should display the enabled checkbox 1`] = `"<CheckBoxControlView label=\\"Auto Bet\\" onChange={[Function]} disabled={false} />"`;
